streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -996,18 +996,19 @@ private void populatePartitionsByHostMaps(final Map<HostInfo, Set<TopicPartition
                 log.debug("Requested client {} to schedule a followup rebalance", clientId);
             }
 
-            log.info("Client {} per-consumer assignment:\n" +
-                "\tprev owned active {}\n" +
-                "\tprev owned standby {}\n" +
-                "\tassigned active {}\n" +
-                "\trevoking active {}" +
-                "\tassigned standby {}\n",
-                clientId,
-                clientMetadata.state.prevOwnedActiveTasksByConsumer(),
-                clientMetadata.state.prevOwnedStandbyByConsumer(),
-                clientMetadata.state.assignedActiveTasksByConsumer(),
-                clientMetadata.state.revokingActiveTasksByConsumer(),
-                clientMetadata.state.assignedStandbyTasksByConsumer());
+            log.info("Client {} per-consumer assignment\n {}",
+                     clientId,
+                     clientMetadata.state.assignedTasksConsumerStateByConsumer()
+                                         .entrySet()
+                                         .stream()
+                                         .map(e -> "\tconsumer : " + e.getKey() + '\n' +
+                                                   e.getValue().entrySet()
+                                                    .stream()
+                                                    .map(taskIdEntry -> "\ttaskId : " + taskIdEntry.getKey()
+                                                                        + " states : " + taskIdEntry.getValue())
+                                                    .collect(Collectors.joining("\n")))
+                                         .collect(Collectors.joining("\n"))
+            );
```

Comment:
This will print everything out with one task per line, right? That might be a problem (or at least, an annoyance) in apps with hundreds or more tasks in total. Also, just speaking from experience in debugging an application, it's more useful to present the information in terms of "here is all the assigned active tasks, here are the revoking active, etc..." just so you can see all the tasks of each type together. So we may want to keep the original format, or something similar at least